feat: added crc support

onmax opened the pull request:
Also added support for smsApiToken in the `/settle` request

sisou commented on OasisApi.ts:
```diff
@@ -360,11 +367,13 @@ function convertHtlc<TStatus extends HtlcStatus>(htlc: RawHtlc<TStatus>): Htlc<T
     switch (asset) {
-        case OasisAsset.EUR: decimals = 2; break;
+        case OasisAsset.EUR:
+        case OasisAsset.CRC:
+            decimals = 2; break;
```
Are we going to handle CRC with 2 decimals? AFAIK decimals in CRC are no longer used. Does Fastspot or OASIS ever use decimals for CRC?

onmax replied:
We won't use decimals, but in the UI we might add two decimals with 2 zeros like `5700.00`

sisou replied:
Just know that this means that the library will return 500 CRC as `50000`, as it converts what the API gives to the smallest unit, and you are specifying here that the smallest unit is 2 decimals lower than the decimal dot.

sisou commented on OasisApi.ts:
```diff
@@ -252,6 +254,7 @@ export async function settleHtlc(
     secret: string,
     settlementJWS: string,
     authorizationToken?: string,
+    smsApiToken?: string,
```
Hmm. This will become a longer and longer list of arguments, if ever another service uses a different token type. And with the CRC OASIS API not taking an `authorizationToken`, the 4th argument to this function will always have to be set to `undefined`. I'd say we should change the tokens into an options object: 

```ts
tokens?: Partial<{
    authorizationToken: string,
    smsApiToken: string,
}>
```

sisou replied:
Also, when changed, this function signature has to be updated in `libswap` OASIS handlers' `OasisClient` interface.

onmax replied:
I was thinking the same, but changing the param to an object it is a breaking change.

Do we want to be backwards compatible?

sisou replied:
I'd say no. We should be the only ones using it, especially since OASIS EUR was shut down for a long time now.

onmax committed:
feat: added crc support

Also added support for smsApiToken in the `/settle` request

sisou committed:
Generalize tokens

onmax committed:
chore: updated bruno collection

Also added post scripts to request to automatically update env variables after each call

onmax committed:
chore: added sinpe body request

sisou commented on bruno_collection/Create HTLC.bru:
```diff
@@ -26,7 +26,13 @@ body:json {
+
+script:post-response {
+  let data = res.getBody();
+  bru.setEnvVar("htlc_id",data.id);
```
Oh, super cool, didn't know this is possible! 👍👍👍
